resharper/resharper-unity/src/CSharp/Feature/OnlineHelp/UnityOnlineHelpProvider.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using JetBrains.Application;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.OnlineHelp;
+using JetBrains.ReSharper.Plugins.Unity.Application.UI.Help;
+using JetBrains.ReSharper.Psi;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.OnlineHelp
+{
+    [ShellComponent]
+    public class UnityOnlineHelpProvider : IOnlineHelpProvider
+    {
+        private readonly ShowUnityHelp myShowUnityHelp;
+
+        public UnityOnlineHelpProvider(ShowUnityHelp showUnityHelp)
+        {
+            myShowUnityHelp = showUnityHelp;
+        }
+
+        public Uri GetUrl(IDeclaredElement element)
+        {
+            if (!IsAvailable(element)) return null;
+            var unityApi = element.GetSolution().GetComponent<UnityApi>();
+            var name = element.GetUnityEventFunctionName(unityApi);
+            
+            if (!ShowUnityHelp.IsUnityKeyword(name)) return null;
+            var keyword = ShowUnityHelp.StripPrefix(name);
```

Comment:
It's a bit weird that we have this implementation split across two classes - they're doing the same thing, why do we need two classes? Does ReSharper still use `IShowHelp`? Could it use `IOnlineHelpProvider` instead? Or provide a default implementation that uses `IOnlineHelpProvider`?

Reply:
I haven't found a way unify IShowHelp and IOnlineHelpProvider.
IShowHelp is coupled with QuickDoc for Re# in VS and IOnlineHelpProvider goes together with ContextHelpActionHandler , which is used with F1 (External Documentation) in VS/Rider.